spec/graphql/schema/warden_spec.rb

```diff
@@ -524,6 +534,15 @@ class C < A; end
 
         class BagOfThings < GraphQL::Schema::Union
           possible_types A, B, C
+
+          if GraphQL::Schema.use_schema_subset?
+            def self.visible?(ctx)
+              (
+                possible_types.any? { |pt| ctx.schema.visible?(pt, ctx) } ||
+                ctx.schema.extra_types.include?(self)
+              ) && super
+            end
+          end
```

Comment:
This is the possible_types thing, where Subset doesn't know if the possible types are hidden or not.